stdlib/public/core/Collection.swift

```diff
@@ -229,8 +229,7 @@ extension CollectionType where SubSequence == Self {
   /// - Complexity: O(1)
   @warn_unused_result
   public mutating func popFirst() -> Generator.Element? {
-    guard !isEmpty else { return nil }
-    let element = first!
+    guard let element = first where !isEmpty else { return nil }
```

Comment:
I think this actually makes the function more expensive, since it evaluates both `first` and `isEmpty`.